CMakeLists.txt

```
@@ -55,11 +53,17 @@ option(DLAF_BUILD_DOC "Build documentation" OFF)
 option(DLAF_WITH_PRECOMPILED_HEADERS "Use precompiled headers." OFF)
 option(DLAF_WITH_SCALAPACK "Build ScaLAPACK-like C API (requires ScaLAPACK)" OFF)
 
-if(DLAF_WITH_MKL AND DLAF_WITH_MKL_LEGACY)
-  message(SEND_ERROR "DLAF_WITH_MKL and DLAF_WITH_MKL_LEGACY are mutually exclusive.")
-endif()
+try_compile(
+  DLAF_WITH_MKL_TRY_COMPILE "${PROJECT_SOURCE_DIR}/cmake/tests"
```

```suggestion
  DLAF_WITH_MKL_TRY_COMPILE "${PROJECT_BINARY_DIR}/cmake/tests"
```

Otherwise it will end up polluting the source dir (we have plenty of rules in `.gitignore` so we don't clutter up the repo anyway).

---

Good catch! That was never meant to be in the source directory...